Tasks/Common/dotNetCliWrapper.ts

```diff
@@ -112,27 +111,94 @@ export class DotNetCliWrapper {
         args.push('--disable-interactive')
         args.push('true')
 
-        return this.executeAsync(args, additionalArgs)
+        return await this.execute(args, additionalArgs)
     }
 
-    public async executeAsync(args: string[], additionalArgs: string): Promise<void> {
-        const dotnetPath = tl.which('dotnet', true)
-        console.log('Path to tool: ' + dotnetPath)
-
-        const dotnet = tl.tool(dotnetPath)
+    public async execute(args: string[], additionalArgs: string, additionalCommandLineOptions?: any): Promise<number> {
+        const dotnet = tl.tool(this.dotnetCliPath)
 
         for (const arg of args) {
             dotnet.arg(arg)
         }
 
         dotnet.line(additionalArgs)
 
-        const execOptions = {
+        const execOptions: any = {
             cwd: this.cwd,
             env: this.env
         }
 
+        if (additionalCommandLineOptions) {
+            // tslint:disable-next-line: no-unsafe-any
+            for (const key of Object.keys(additionalCommandLineOptions)) {
+                // tslint:disable-next-line: no-unsafe-any
+                execOptions.key = additionalCommandLineOptions[key]
+            }
+        }
+
         // tslint:disable-next-line: no-unsafe-any
-        await dotnet.exec(execOptions as any)
+        return await dotnet.exec(execOptions)
+    }
+
+    private async restore(): Promise<number> {
+        return await this.execute(['restore'], '')
+    }
+
+    private async checkForGlobalLambdaToolsInstalled(): Promise<boolean> {
+        try {
+            const returnCode = await this.execute(['lambda', 'help'], '', { silent: true })
+            if (returnCode !== 0) {
+                return false
+            }
+        } catch (exception) {
+            return false
+        }
+
+        return true
+    }
+
+    private async installGlobalTools(): Promise<boolean> {
+        try {
+            const returnCode = await this.execute(['tool', 'install', '-g', 'Amazon.Lambda.Tools'], '')
+            if (returnCode === 0) {
+                return true
+            }
+        } catch (e) {}
+
+        // If something went wrong in the last step, we try to update the tools instead
+        try {
+            const returnCode = await this.execute(['tool', 'update', '-g', 'Amazon.Lambda.Tools'], '')
+            if (returnCode === 0) {
+                return true
+            }
+        } catch (e2) {
+            return false
```

Comment:
To help diagnose execution issues, consider logging the error and when the return code isn't 0 if this isn't happening already.

Reply:
printing the error now